packages/service-library/tests/rabbitmq/test_rabbitmq.py

```diff
@@ -303,6 +303,71 @@ async def _always_returning_fail(_: Any) -> bool:
     }
 
 
+@pytest.mark.parametrize("topics", _TOPICS)
+@pytest.mark.no_cleanup_check_rabbitmq_server_has_no_errors()
+async def test_publish_with_no_registered_subscriber(
+    on_message_spy: mock.Mock,
+    create_rabbitmq_client: Callable[[str], RabbitMQClient],
+    random_exchange_name: Callable[[], str],
+    random_rabbit_message: Callable[..., PytestRabbitMessage],
+    mocked_message_parser: mock.AsyncMock,
+    topics: list[str] | None,
+):
+    publisher = create_rabbitmq_client("publisher")
+    consumer = create_rabbitmq_client("consumer")
+
+    exchange_name = f"{random_exchange_name()}"
+
+    ttl_s: float = 0.1
+    topics_count: int = 1 if topics is None else len(topics)
+
+    async def _publish_random_message():
+        if topics is not None:
+            for topic in topics:
+                message = random_rabbit_message(topic=topic)
+                await publisher.publish(exchange_name, message)
+        else:
+            message = random_rabbit_message()
+            await publisher.publish(exchange_name, message)
+
+    async def _subscribe_consumer_to_queue():
+        await consumer.subscribe(
+            exchange_name,
+            mocked_message_parser,
+            topics=topics,
+            exclusive_queue=False,
+            message_ttl=int(ttl_s * 1000),
```

Comment:
random THOUGHT: sometimes I think it might be worth using  https://pint.readthedocs.io/en/stable/ to get units right. Here is ms right? I guess here is an overkill but who knows

For example, in **the very same interface y**ou have `message_ttl` in ms and `unexpected_error_retry_delay_s` in seconds.

Reply:
Ohh boy here it would be a huge overkill. But it's a good one to keep in mind.
Usually we use `timedelta` and `ByteSize` where it makes sense (interfaces).

Here I wanted to migrate to `timedelta`, but it does not have a millisecond resolution. It's better left like this.